test/examples/test_delivery_mechanisms_isolated.py

```diff
@@ -0,0 +1,154 @@
+# Copyright 2024 Proyectos y Sistemas de Mantenimiento SL (eProsima).
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import subprocess
+import pytest
+
+isolated_delivery_test_cases = [
+    # Default builtin transports (UDP forced)
+    ('', '-s 20', '-s 20', '', 50),
+    ('', '-s 20', '-s 20', '--ignore-local-endpoints', 50),
+    # Data-sharing only in the same container
+    ('--mechanism data-sharing', '--mechanism data-sharing', '--unknown-argument', '--unknown-argument', 10),
```

Comment:
Maybe these two are not strictly necessary since we already test it in the other tests
```suggestion
```